feathr_project/feathr/api.py

```diff
@@ -0,0 +1,154 @@
+    """
+    Defining contract for input field
+    """
+
+    features: List[str]
+
+
+@app.get("/projects/{project_name}/features", response_model=Features)
+def list_registered_features(code, project_name: str, response: Response):
+    """List all the already registered features. If project_name is not provided or is None, it will return all
+    the registered features; otherwise it will only return features under this project
+    """
+    if code != appServiceKey:
+        raise HTTPException(status_code=403, detail="You are not allowed to access this resource")  
+    try:
+        registry_client = getRegistryClient()
+        logger.info("Retrieved registry client successfully")
+        response.status_code = status.HTTP_200_OK
+        result = registry_client.list_registered_features(project_name)
+        return {"features" : result}
+    except AtlasException as ae:
+        logger.error("Error retrieving feature: %s", ae.args[0])
+        raise HTTPException(status_code=400, detail="Error: " + ae.args[0])
+    except Exception as err:
+        logger.error("Error: %s", err.args[0])
+        raise HTTPException(status_code=400, detail="Error: " + err.args[0])
+
+
+@app.get("/projects/{project_name}/features/{feature_name}")
+def get_feature_qualifiedName(code : str, project_name: str, feature_name: str, response: Response, type_name: Optional[str] = None):
+    """List all the already registered features. If project_name is not provided or is None, it will return all
+    the registered features; otherwise it will only return features under this project
+    """
+    if code != appServiceKey:
+        raise HTTPException(status_code=403, detail="You are not allowed to access this resource")  
+    try:
+        registry_client = getRegistryClient()
+        logger.info("Retrieved registry client successfully")
+        response.status_code = status.HTTP_200_OK
+        result = None
+        if type_name: # Type is provided
+            result = registry_client.get_feature_by_fqdn_type(feature_name, type_name)
+        else:
+            result = registry_client.get_feature_by_fqdn(feature_name)
+        return result
+    except AtlasException as ae:
+        logger.error("Error retrieving feature: %s", ae.args[0])
+        raise HTTPException(status_code=400, detail=ae.args[0])
+    except Exception as err:
+        logger.error("Error: %s", err.args[0])
+        raise HTTPException(status_code=400, detail=err.args[0])
+
+
+@app.get("/projects/{project_name}/features/lineage/{feature_name}")
+def get_feature_qualifiedName(code : str, project_name: str, feature_name: str, response: Response):
+    """List all the already registered features. If project_name is not provided or is None, it will return all
+    the registered features; otherwise it will only return features under this project
+    """
+    if code != appServiceKey:
+        raise HTTPException(status_code=403, detail="You are not allowed to access this resource")  
```

Comment:
Shall we have more explicit detail error message? "Not allowed to access" might be confusing.

Reply:
This is being handled in a separate PR by Yihui,thanks for the feedback.